core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -4674,6 +4675,9 @@ private static class CorrelationCollector extends RelHomogeneousShuttle {
     private final VariableUsedVisitor vuv = new VariableUsedVisitor(this);
 
     @Override public RelNode visit(RelNode other) {
+      if (other instanceof HepRelVertex) {
```

Comment:
You might like to cherry-pick in my 5fb70e196ddafaceaf1349378de88e48f9f232f3 commit, which adds the `RelNode.stripped()`. With that commit, this `if` might not be required.

Reply:
The `RelOptUtil` class remains unchanged in the latest patch so no need to cherry-pick anything.